elastic/beats
# Add OpenMetrics Metricbeat module

## What does this PR do?

This PR adds OpenMetrics Metricbeat module as a light module on top of Prometheus collector metricset.

## Why is it important?

OpenMetrics is an effort to standardize exposing metric data based on the Prometheus exposition format. This standalone module will provide an easy and out of the box way to collect metrics from endpoints that follow this format. 

## Checklist


## How to test this PR locally
1. One can use a nodexporter as an example Prometheus app to collect from. For instance this forked project should be enough: https://github.com/ChrsMark/dockprom. Download and `docker-compose up`. Then metrics should be exported at `http://localhost:9100/metrics`.
2. Configuration:
```
- module: openmetrics
  metricsets: ['collector']
  period: 10s
  hosts: ['localhost:9100']

  # This module uses the Prometheus collector metricset, all
  # the options for this metricset are also available here.
  metrics_path: /metrics
  metrics_filters:
    include: ["node_filesystem_*"]
    exclude: ["node_filesystem_device_*", "^node_filesystem_readonly$"]
```
And expect to see only events that match `node_filesystem_*` but also not `node_filesystem_device_*` and are not `node_filesystem_readonly`. 
3. Try to remove `^node_filesystem_readonly$` from `exclude` filters and see that this metric is included in events. One can try different combinations.

## Related issues

- Relates https://github.com/elastic/beats/issues/14982
